### ISSUE TITLE
Export of CGMES Diagram Layout data (DL file)

### ISSUE THREAD

**Issue by sylvlecl** [powsybl/powsybl-incubator#28]
* **Do you want to request a *feature* or report a *bug*?**

Feature: the import of DL data into IIDM extensions to be used for the display of voltage levels is implemented. In the future, we will need to be able to create ourselves those DL files.

* **What is the current behavior?**

No export of layout information as DL file.

* **What is the expected behavior?**

We should be able to export a DL file together with other CGMES data. This can be done provided we have the necessary data in IIDM extensions. 2 different use cases would be :  
- DL file --> IIDM extensions --> DL file
- no layout info --> compute layout info --> IIDM extensions --> DL file

* **What is the motivation / use case for changing the behavior?**

Being able to provide layout data to third party tools, together with network model.

**Comment by massimo-ferraro** [comment]
An issue to be tackled for implementing export of CGMES DL data is that the layout info computed automatically (e.g. from an XIIDM network) has the coordinates of the network components referred to the voltage level, while the CGMES DL profile (at least in the examples I used) have coordinates referred to the whole network (some components, e.g. branches, actually belong to different voltage levels). So, some way to combine the different voltage level layouts could/would be necessary

**Comment by geofjamg** [comment]
There is an ongoing dev for having a hierarchical 3 layers system layout (voltage level -> substation -> zone). It is clearly needed to be able to generate a compliant DL profile.
In the meantime I propose in parallel of 3 layers layout dev to start generating a DL profile with all voltage levels, not connected each other and displayed in kind of grid (alpha order).

**Comment by geofjamg** [comment]
To complete this issue we should  be able to display a "zone" diagram i.e any subset of substation
We could start a new PR that add a "ZoneDiagramLayout"  that will be a layout on top of SubstationLayout and a CGMES implementation.

**Comment by massimo-ferraro** [comment]
There is no modeling of zones in IIDM. Should we define a "zone" as a list of substations (e.g. having List<Substation> as input for the ZoneDiagramLayout), or some more complex modeling is needed?

**Comment by geofjamg** [comment]
> 
> 
> There is no modeling of zones in IIDM. Should we define a "zone" as a list of substations (e.g. having List as input for the ZoneDiagramLayout), or some more complex modeling is needed?

A list of substation is ok.

**Comment by geofjamg** [comment]
We continue this issue in #190
